src/lib.rs

```diff
@@ -101,6 +103,19 @@ pub struct GlobalCrateId {
     pub disambiguator: (u64, u64),
 }
 
+/// Where a crate came from on the local filesystem. One of these three options
+/// must be non-None.
+///
+/// Mimicks CrateSource from src/librustc/middle/cstore.rs (only paths)
+#[cfg_attr(feature = "serialize-serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serialize-rustc", derive(RustcDecodable, RustcEncodable))]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CrateSource {
+    pub dylib: Option<PathBuf>,
+    pub rlib: Option<PathBuf>,
+    pub rmeta: Option<PathBuf>,
+}
```

Comment:
I know we discussed this, but I'm not sure if we got to any kind of conclusion. IIRC, there's only ever multiple sources for std and we only want this info to avoid parsing the command line, but even then it would be better to use the crate name rather than the path. Is that right? Is there anything we need to make the disctinction between these kinds of crate sources?

Reply:
The main use for this was to figure out the dependencies between `rustc` compilations, combined with the new `output` field, so we can reconstruct file-based input-output 'white boxes' in the build plan.

However, we might just discard this information completely since crate disambiguator (included for the local crate and deps) information is sufficient to figure out the dependency relation between crates and by adding the `output` field we also get the implicit file-based dependencies.

I'll remove the struct in the next commit.